benchmark/ext_helpers/daal4py.py

```diff
@@ -37,7 +42,15 @@ def prepare_fit(self, X, y=None, sample_weight=None):
         return super().prepare_fit(X, y, sample_weight)
 
     def init_centroids(self, X):
-        return super(KMeansEngine, self).init_centroids(X)
```

Comment:
This is a very important bugfix, I will open in a separate PR to fasttrack.

Basically, currently on main `k-means++` benchmark with the daal actually runs sklearn's. After fixing, we can see that the daal is more that 10 times faster than sklearn at kmeans++.

Reply:
It would be very interesting to repriorize the optimization of kmeans++, both algorithmically (via subsampling as suggested by @jeremiedbb elsewhere) and with efficient Cython code for the computation of the argmin of distances if it's not already the case.